examples/etcd/etcd-in-cluster/etcd-incluster-blueprint.yaml

```diff
@@ -0,0 +1,97 @@
+apiVersion: cr.kanister.io/v1alpha1
+kind: Blueprint
+metadata:
+  name: etcd-blueprint
+  namespace: kanister
+actions:
+  backup:
+    outputArtifacts:
+      cloudObject:
+        keyValue:
+          backupLocation: "{{ .Phases.uploadSnapshot.Output.backupLocation }}"
+    phases:
+    - func: KubeExec
+      name: takeSnapshot
+      objects:
+        etcdConfig:
+          kind: Secret
+          name: "etcd-{{ .Object.metadata.namespace }}"
+          namespace: "{{ .Object.metadata.namespace }}"
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        pod: "{{ .Object.metadata.name }}"
+        container: etcd
+        command:
+          - sh
+          - -c
+          - |
+            export cacert="{{ index .Phases.takeSnapshot.Secrets.etcdConfig.Data "cacert" | toString }}"
+            export cert="{{ index .Phases.takeSnapshot.Secrets.etcdConfig.Data "cert" | toString }}"
+            export endpoints="{{ index .Phases.takeSnapshot.Secrets.etcdConfig.Data "endpoints" | toString }}"
+            export key="{{ index .Phases.takeSnapshot.Secrets.etcdConfig.Data "key" | toString }}"
+
+            ETCDCTL_API=3 etcdctl --endpoints=$endpoints --cacert=$cacert  --cert=$cert  --key=$key snapshot save /tmp/etcd-backup.db
+            ETCDCTL_API=3 etcdctl --endpoints=$endpoints --cacert=$cacert  --cert=$cert  --key=$key snapshot status /tmp/etcd-backup.db
+
+    - func: KubeTask
+      name: uploadSnapshot
+      args:
+        namespace: "kanister"
+        image: kanisterio/kanister-kubectl:1.18
+        command:
+          - sh
+          - -c
+          - |
+            BACKUP_LOCATION=etcd_backups/{{ .Object.metadata.namespace }}/{{ .Object.metadata.name }}/{{ toDate "2006-01-02T15:04:05.999999999Z07:00" .Time | date "2006-01-02T15:04:05Z07:00" }}/etcd-backup.db.gz
+            kubectl cp {{ .Object.metadata.namespace }}/{{ .Object.metadata.name }}:/tmp/etcd-backup.db /tmp/etcd-backup.db
+            gzip /tmp/etcd-backup.db
+            kando location push --profile '{{ toJson .Profile }}'  /tmp/etcd-backup.db.gz --path $BACKUP_LOCATION
+            kando output backupLocation $BACKUP_LOCATION
+
+    - func: KubeExec
+      name: removeSnapshot
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        pod: "{{ .Object.metadata.name }}"
+        container: etcd
+        command:
+          - sh
+          - -c
+          - |
+            rm -rf  /tmp/etcd-backup.db
+
+  restore:
+    inputArtifactNames:
+    - cloudObject
+    phases:
+    - func: KubeExec
+      name: showRestoreHelp
+      args:
+        namespace: "{{ .Object.metadata.namespace }}"
+        pod: "{{ .Object.metadata.name }}"
+        container: etcd
+        command:
+          - sh
+          - -c
+          - |
+            echo "Automated restore for ETCD is not supported please follow the examples docs to restore the backup manually. Backup path : "{{ .ArtifactsIn.cloudObject.KeyValue.backupLocation }}
```

Comment:
Instead of printing log, we can just set backuppath as an output artifact so that the user can get the info by describing the restore actionset.

Reply:
@PrasadG193 
yeah, I have done that. If you see the new steps that I have included. That explains to describe the actionset to get the backup path.